lib/cisco_node_utils/ace.rb

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # Ace - node utility class for Ace Configuration
+  class Ace < NodeUtil
+    attr_reader :acl_name, :seqno, :afi, :set_args, :get_args
+    attr_reader :regexp_str
+
+    # acl_name is name of acl
+    # seqno is sequence number of ace
+    # afi is either v4 or v6
+    def initialize(acl_name, seqno, afi)
+      fail TypeError unless
+        acl_name.is_a?(String) && seqno.is_a?(Integer) && afi.is_a?(String)
+      fail ArgumentError 'we expect ip or ipv6' unless
+        afi == 'ip' || afi == 'ipv6'
+      @acl_name = acl_name
+      @afi = afi
+      @seqno = seqno
+      @set_args = @get_args = { acl_name: @acl_name, seqno: @seqno, afi: @afi,
+                                src_port: '', dst_port: '',
+                                option_format: '' }
+      @regexp_str = '(?<seqno>\d+) (?<action>\S+)'\
+                 ' *(?<proto>\d+|\S+)'\
+                 ' *(?<src_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)*'\
+                 ' *(?<src_port>eq \S+|neq \S+|lt \S+|''gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<dst_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)'\
+                 ' *(?<dst_port>eq \S+|neq \S+|lt \S+|gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<option>[a-zA-Z0-9-\/ ]*)*'
+    end
+
+    # Create a hash of all aces under give acl_name.
+    def self.aces
+      afis = %w(ip ipv6)
+      hash = {}
+      afis.each do |afi|
+        @get_args = { afi: afi }
+        instances = config_get('acl', 'all_acl', @get_args)
+        next if instances.nil?
+        instances.each do |name|
+          hash[name] = Acl.new(name, @get_args[:afi], false)
```

Comment:
Why are you creating new Acl objects in this method?  This method is for collecting Ace instances.